src/safeposix/syscalls/fs_calls.rs

```diff
@@ -71,6 +71,9 @@ impl Cage {
     *   mkdir() will return 0 when success and -1 when fail 
     */
     pub fn mkdir_syscall(&self, path: &str, mode: u32) -> i32 {
+        if (mode & !0o777) != 0 {
+            return syscall_error(Errno::EPERM, "mkdir", "Invalid mode bits specified");
+        }
```

Comment:
what is currently rawposix returned in this situation?

Reply:
Could you test with [`get_errno()`](https://github.com/Lind-Project/RawPOSIX/blob/98fedeab93440d63b69e044ca81a3da5f84220d5/src/interface/misc.rs#L139) to see the errno returned by libc::mmap?

Sth like:
```rust
let ret = libc::mmap(...)
if ret < 0 {
  println!(get_errno());
}
```